lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -837,7 +837,7 @@ def post(self):
         # Automatically authorize the operation only if the API schema has the
         # "user" key; otherwise we assume that authorization is unnecessary, or
         # that the API-specific subclass will take care of that in preprocess.
-        if "user" in self.schema:
+        if "user" in json_data:
```

Comment:
It's not correct to make the authorization test only if someone specifies `"user"` in the request payload. `"user"` must be a defined Schema keyword with type USER. If you follow through my current PRs, it's evolved to that point in #2417, although main is currently more simplistic.

If the caller omits `"user"` (where it's optional), we *still* need to run `_check_authorization` against the `"access"` value; e.g., we need to fail if they specify `{"access": "private"}` and there's no authorized user or it isn't ADMIN.

Reply:
Ah right we need to check the access as well.